tools/dv/samples/kmdf/pool_leak/src/lib.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// License: MIT OR Apache-2.0
+
+//! # Abstract
+//!
+//!     This KMDF sample contains an intentional error that is designed to
+//! demonstrate the capabilities and features of Driver Verifier and the Device
+//! Fundamental tests.
+//!     
+//!     The driver is desgined to allocate memory using ExAllocatePool2 to its
+//! Device Context buffer when a device is added by the PnP manager. However,
+//! this buffer is not freed anywhere in the driver, including the driver unload
+//! function.
+//!
+//!     By enabling the Driver Verifier on this driver, the pool leak
+//! violation can be caught when the driver is unloaded and with an active KDNET
+//! session, the bug can be analyzed further.
+
+#![no_std]
+#![cfg_attr(feature = "nightly", feature(hint_must_use))]
+#![deny(clippy::all)]
+#![warn(clippy::pedantic)]
+#![warn(clippy::nursery)]
+#![warn(clippy::cargo)]
+#![allow(clippy::missing_safety_doc)]
```

Comment:
Clippy complains about CamelCase entries in docs when they don't have backticks around them (i.e. `MyFunction`.)  You _should_ go and add backticks to function names in your documentation name, but we need to add a suppression here to stop it from complaining about "PnP".

```suggestion
#![allow(clippy::missing_safety_doc)]
#![allow(clippy::doc_markdown)]
```